rusqlite_migration/src/loader.rs

```diff
@@ -0,0 +1,80 @@
+use std::collections::BTreeMap;
+
+use crate::{Error, Result, M};
+use include_dir::{Dir, DirEntry};
+use log::trace;
+
+struct MigrationFile {
+    up: &'static str,
+    down: Option<&'static str>,
+}
+
+pub fn from_dir(dir: &'static Dir<'static>) -> Result<Vec<M<'static>>> {
+    let mut migrations: BTreeMap<usize, MigrationFile> = BTreeMap::new();
+
+    let entries = dir
+        .find("*.sql")
+        .map_err(|_| Error::FileLoad("Incorrect glob pattern".to_string()))?
+        .flat_map(|entry| match entry {
+            DirEntry::File(f) => Some(f),
+            DirEntry::Dir(_) => None,
```

Comment:
We don’t want to go into sub directories, that makes sense. Perhaps this should be documented in the top caller (`M::from_directory`).

Reply:
Good catch!